creator/projects/schema.py

```diff
@@ -0,0 +1,164 @@
+        User must be authenticated and be ADMIN.
+        """
+        user = info.context.user
+        if user is None or not user.is_authenticated:
+            raise GraphQLError("Not authenticated to mutate a Cavatica project.")
+
+        try:
+            cavaticaProject = CavaticaProject.objects.get(project_id=project_id)
+        except CavaticaProject.DoesNotExist:
+            raise GraphQLError("Cavatica project does not exist.")
+
+        if "ADMIN" not in user.ego_roles:
+            raise GraphQLError("Not authenticated to mutate a Cavatica project.")
+
+        try:
+            if kwargs.get("name"):
+                cavaticaProject.state = kwargs.get("name")
+            if kwargs.get("description"):
+                cavaticaProject.description = kwargs.get("description")
+            cavaticaProject.save()
+        except ClientError:
+            raise GraphQLError("Failed to save version mutation.")
+
+        return CavaticaProjectMutation(cavaticaProject=cavaticaProject)
+
+
+class CreateCavaticaProjectMutation(graphene.Mutation):
+    class Arguments:
+        project_id = graphene.String(
+            description='The Cavatica project identifier',
+            required=False
+        )
+        name = graphene.String(
+            description='The name of the Cavatica project',
+            required=True
+        )
+        url = graphene.String(
+            description='The url of the Cavatica project',
+            required=True
+        )
+        project_type = graphene.String(
+            description='The Cavatica project type',
+            required=False
+        )
+        description = graphene.String(
+            description='The description markdown of the Cavatica project',
+            required=False
+        )
+
+    success = graphene.Boolean()
+    cavaticaProject = graphene.Field(CavaticaProjectNode)
+
+    def mutate(self, info, file, fileId, description, **kwargs):
+        """
+        Create a new Cavatica project
+        """
+        user = info.context.user
+        if user is None or not user.is_authenticated:
+            raise GraphQLError("Not authenticated to create a Cavatica project")
+
+        if ("ADMIN" not in user.ego_roles):
+            raise GraphQLError("Not authenticated to upload to the study.")
+
+        try:
+            cavaticaProject = CavaticaProject(
+                project_id=project_id,
+                name=name,
+                url=url,
+                project_type=project_type,
+                description=description,
+            )
+            cavaticaProject.save()
+        except ClientError:
+            raise GraphQLError("Failed to save create Cavatica project")
+
+        return CreateCavaticaProjectMutation(success=True, cavaticaProject=cavaticaProject)
+
+
+class CavaticaProjectQuery(object):
+    cavaticaProject = relay.Node.Field(CavaticaProjectNode)
+    cavatica_project_by_project_id = Field(CavaticaProjectNode, project_id=String(required=True))
```

Comment:
I think the proper 'graphQL' way to do this is to use a filter on the `allProjects` query. We can look into this.